pennylane/pulse/parametrized_evolution.py

```diff
@@ -178,18 +171,13 @@ def __init__(
                 "All operators inside the parametrized hamiltonian must have a matrix defined."
             )
         self.H = H
-        self.time = time
         self.params = params
         self.odeint_kwargs = odeint_kwargs
-        if t is None:
-            self.t = None
-        else:
-            self.t = jnp.array([0, t] if qml.math.ndim(t) == 0 else t, dtype=float)
+        self.t = jnp.array(t, dtype=float)
         super().__init__(wires=H.wires, do_queue=do_queue, id=id)
 
-    def __call__(self, params, t, **odeint_kwargs):
+    def __call__(self, params, **odeint_kwargs):
```

Comment:
I think we need to be able to update `t` to modify pulse durations, no? Pulse duration should definitely be a trainable parameter.

Reply:
Or I guess we just pass `t` to the `circuit` to update `t`, so it doesn't matter where we define it, because we call the whole circuit again for each repetition. So maybe never mind?

Reply:
Exactly, moving `t` from this method to `qml.evolve` doesn't affect its trainability.

Reply:
Moved `t` back to `ParametrizedEvolution.__call__`

Reply:
Moved `t` back to `ParametrizedEvolution.__call__`